src/processes/follow-topic-process.js

```diff
@@ -1,7 +1,95 @@
+const CHANNEL_TYPE_ANONYMOUS = require('../utils/constant');
+const updateBetterChannelMembers = require('../getstream/updateBetterChannelMembers');
 
-const {LogError, UserTopic, CommunityMessageFormat} = require('../databases/models');
+const generate_channel_id_for_anon_chat = (owner, member, context = null, source_id = null) => {
+  const hash = crypto.createHash('sha256');
+  hash.update(owner);
+  hash.update(member);
+  if (context) hash.update(context);
+  if (source_id) hash.update(source_id);
 
+  return hash.digest('hex');
+};
+const anon_to_sign_user = async (client, user, targetUser) => {
+  const channel = await ChatAnonUserInfo.findOne({
+    where: {
+      my_anon_user_id: user,
+      target_user_id: targetUser,
+      context: 'auto message',
+      initiator: user
+    }
+  });
+  if (channel) {
+    return channel;
+  } else {
+    const channel_id = generate_channel_id_for_anon_chat(user, targetUser, 'autoMessage');
+    const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+    const color = BetterSocialConstantListUtils.getRandomColor();
+    const anon_init_data = {
+      anon_user_info_color_code: color.code,
+      anon_user_info_color_name: color.color,
+      anon_user_info_emoji_code: emoji.emoji,
+      anon_user_info_emoji_name: emoji.name
+    };
+
+    const new_channel = await ChatAnonUserInfo.create({
+      channel_id: channel_id,
+      my_anon_user_id: user,
+      target_user_id: targetUser,
+      anon_user_info_color_code: anon_init_data.anon_user_info_color_code,
+      anon_user_info_color_name: anon_init_data.anon_user_info_color_name,
+      anon_user_info_emoji_code: anon_init_data.anon_user_info_emoji_code,
+      anon_user_info_emoji_name: anon_init_data.anon_user_info_emoji_name,
+      context: 'auto message',
+      initiator: user,
+      source_id: ''
+    });
+    return new_channel;
+  }
+};
+const sendMessageAsAnonymous = async (serverClient, communityMessageFormat, data) => {
+  const members = [communityMessageFormat.user_id, data.user_id];
+  const channel_info = await anon_to_sign_user(
+    serverClient,
+    communityMessageFormat.user_id,
+    data.user_id
+  );
+
+  const newChannel = serverClient.channel('messaging', channel_info.channel_id, {
+    type_channel: CHANNEL_TYPE_ANONYMOUS,
+    members,
+    created_by_id: communityMessageFormat.user_id
+  });
+  const createdChannel = await newChannel.create();
+  const channelState = await newChannel.watch();
+
+  await updateBetterChannelMembers(newChannel, createdChannel, true, {
+    channel_type: CHANNEL_TYPE_ANONYMOUS
+  });
+
+  if (channelState.messages.length === 0) {
+    await newChannel.sendMessage({
+      text: communityMessageFormat.message,
+      user_id: communityMessageFormat.user_id
+    });
+
+    try {
+      // await chat.stopWatching();
+    } catch (error) {
+      console.log('::: Error on stopWatching :::', error);
+    }
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `sendMessageAsAnonymous` function, you are creating a new channel and sending a message as an anonymous user. However, there is no error handling for the case where any of these operations fail. It would be good to wrap this in a try-catch block and handle any potential errors that might occur during these operations.



<!-- This is an auto-generated comment by OSS CodeRabbit -->